feat: Add comprehensive zero-shot classification example script

rmitsch opened the pull request:
## Description
Adds a comprehensive example script for zero-shot classification using Sieves + Outlines. The script provides a complete workflow for classifying Hugging Face datasets with automatic device detection, structured outputs, and dataset publishing.

## Related Issues
-

## Changes Made
- **New example script**: `examples/create_classification_dataset_with_sieves.py` - A comprehensive uv-compatible script for zero-shot text classification
- **Enhanced Classification task**: Improved `to_hf_dataset()` method with proper label normalization and multi-label support
- **Package metadata**: Updated setup.py with corrected package name

### Key Features of the Example Script:
- **Automatic device detection**: Dynamically selects GPU or CPU based on availability
- **Dual model support**: Handles both dedicated zero-shot classification models and general language models via Outlines
- **Multi-label classification**: Optional multi-label mode with configurable thresholds
- **Comprehensive preprocessing**: Text validation, length limits, and cleaning
- **Rich logging**: Detailed statistics and distribution reporting
- **Label descriptions**: Optional semantic descriptions for better classification
- **HF Hub integration**: Automatic authentication and dataset publishing
- **uv compatibility**: Uses PEP 723 inline script metadata for dependency management

### Usage Examples:
```bash
# Simple binary classification
uv run examples/create_classification_dataset_with_sieves.py \
  --input-dataset stanfordnlp/imdb \
  --column text \
  --labels "positive,negative" \
  --model MoritzLaurer/deberta-v3-large-zeroshot-v2.0 \
  --output-dataset your-username/imdb-classified

# Multi-label with descriptions
uv run examples/create_classification_dataset_with_sieves.py \
  --input-dataset ag_news \
  --column text \
  --labels "world,sports,business,science" \
  --multi-label \
  --model MoritzLaurer/deberta-v3-large-zeroshot-v2.0 \
  --output-dataset your-username/agnews-multilabel
```

## Checklist
- [x] Tests have been extended to cover changes in functionality
- [x] Existing and new tests succeed  
- [x] Documentation updated (if applicable)
- [x] Related issues linked

## Screenshots/Examples (if applicable)
The script provides comprehensive logging output showing:
- Device selection (GPU/CPU)
- Dataset loading and preprocessing statistics
- Classification progress and results
- Label distribution analysis
- Success rates and error handling

ivyleavedtoadflax commented:
@davanstrien is there a way to use a custom docker container?

rmitsch commented:
@ivyleavedtoadflax You can use any image as base AFAIK, as long as it's on the Docker Hub. Maybe give https://hub.docker.com/r/nvidia/cuda a shot?

ivyleavedtoadflax commented:
> @ivyleavedtoadflax You can use any image as base AFAIK, as long as it's on Docker Hub. Maybe give https://hub.docker.com/r/nvidia/cuda a shot?

Also failed. I started looking in to using a custom docker container - the same that HF uses but just with the OpenCV deps added.

Note that this is happening because some of the OCR dependencies require it. If we moved the OCR dependencies to `extra` then we might be able to sidestep the problem entirely

rmitsch commented:
> Also failed. I started looking in to using a custom docker container - the same that HF uses but just with the OpenCV deps added.
> 
> Note that this is happening because some of the OCR dependencies require it. If we moved the OCR dependencies to extra > then we might be able to sidestep the problem entirely
@ivyleavedtoadflax Done in https://github.com/MantisAI/sieves/pull/163. Also updated to the new API design. Try again? Shouldn't require `opencv` anymore and work with the default Docker image.
